mmdet/structures/bbox/horizontal_bbox.py

```diff
@@ -0,0 +1,324 @@
+        (x2, y2)) to (x1, y1, x2, y2).
+
+        Args:
+            corners (Tensor): Corner tensor with shape of (..., 4, 2).
+
+        Returns:
+            Tensor: Horizontal box tensor with shape of (..., 4).
+        """
+        min_xy = corners.min(dim=-2)[0]
+        max_xy = corners.max(dim=-2)[0]
+        return torch.cat([min_xy, max_xy], dim=-1)
+
+    def rescale_(self,
+                 scale_factor: Tuple[float, float],
+                 mapping_back=False) -> None:
+        """Inplace rescale boxes w.r.t. rescale_factor.
+
+        Note:
+            Both ``rescale_`` and ``resize_`` will enlarge or shrink bboxes
+            w.r.t ``scale_facotr``. The difference is that ``resize_`` only
+            changes the width and the height of bboxes, but ``rescale_`` also
+            rescales the box centers simultaneously.
+
+        Args:
+            scale_factor (Tuple[float, float]): factors for scaling boxes.
+                The length should be 2.
+            mapping_back (bool): Mapping back the rescaled bboxes.
+                Defaults to False.
+        """
+        bboxes = self.tensor
+        assert len(scale_factor) == 2
+        scale_factor = bboxes.new_tensor(scale_factor).repeat(2)
+        self.tensor = bboxes / scale_factor if mapping_back else \
+            bboxes * scale_factor
+
+    def resize_(self, scale_factor: Tuple[float, float]) -> None:
+        """Inplace resize the box width and height w.r.t scale_factor.
+
+        Note:
+            Both ``rescale_`` and ``resize_`` will enlarge or shrink bboxes
+            w.r.t ``scale_facotr``. The difference is that ``resize_`` only
+            changes the width and the height of bboxes, but ``rescale_`` also
+            rescales the box centers simultaneously.
+
+        Args:
+            scale_factor (Tuple[float, float]): factors for scaling box
+                shapes. The length should be 2.
+        """
+        bboxes = self.tensor
+        assert len(scale_factor) == 2
+        ctrs = (bboxes[..., 2:] + bboxes[..., :2]) / 2
+        wh = bboxes[..., 2:] - bboxes[..., :2]
+        scale_factor = bboxes.new_tensor(scale_factor)
+        wh = wh * scale_factor
+        xy1 = ctrs - 0.5 * wh
+        xy2 = ctrs + 0.5 * wh
+        self.tensor = torch.cat([xy1, xy2], dim=-1)
+
+    def is_bboxes_inside(self, img_shape: Tuple[int, int]) -> BoolTensor:
+        """Find bboxes inside the image.
+
+        In ``HorizontalBoxes``, as long as a part of the box is inside the
+        image, this box will be regarded as True.
+
+        Args:
+            img_shape (Tuple[int, int]): A tuple of image height and width.
+
+        Returns:
+            BoolTensor: Index of the remaining bboxes. Assuming the original
+            horizontal boxes have shape (m, n, 4), the output has shape (m, n).
+        """
+        img_h, img_w = img_shape
+        bboxes = self.tensor
+        return (bboxes[..., 0] < img_w) & (bboxes[..., 2] > 0) \
```

Comment:
`bboxes[..., 2] > 0` or `bboxes[..., 2] >= 0`?

Reply:
It is moved from `find_inside_bboxes`, where the original setting is `bboxes[..., 2] > 0`.

Reply:
A more important reason I think is to ensure the selected boxes have large than zero areas inside the image. I will add more docstring in this function.